### ISSUE TITLE
The lambda finishes before generating the output

### ISSUE THREAD

**Issue by yabrego** [binoculars/aws-lambda-ffmpeg#48]
Hello!

I've tried to run the lambda, as is but I'm having trouble figuring out why it finishes executing early.
I'm running it with the example ffmpeg command if I run this command locally on my machine it works, on the lambda however it appears to never execute.

Below is the output that I can see in the logs.

```
2017-05-25T19:56:31.061Z	3f011341-1234-123e-1234-xxxxxxxx	Valid video stream found. FFprobe finished.
2017-05-25T19:56:31.061Z	3f011341-1234-123e-1234-xxxxxxxx	Starting FFmpeg
2017-05-25T19:56:31.079Z	3f011341-1234-123e-1234-xxxxxxxx	Deleting /tmp/download
END RequestId: 3f011341-1234-123e-1234-xxxxxxxx
REPORT RequestId: 3f011341-1234-123e-1234-xxxxxxxx	Duration: 376.91 ms	Billed Duration: 400 ms Memory Size: 1536 MB	Max Memory Used: 124 MB
```
It looks like the /tmp/output directory is always empty, so I feel like there is something failing in the FFMPEG execution itself, I have tried with very short videos, to see if it was an issue of storage or something like that, but it looks like that is not the case.

I'm also using 3 buckets, one for the source file, one for the lambda and one for the destination.
I have in the lambda's environment variables, 
```
DESTINATION_BUCKET=mybucket
VIDEO_MAX_DURATION=180
USE_GZIP=false
MIME_TYPES={"png":"image/png","mp4":"video/mp4"}
FFMPEG_ARGS=-c:a copy -vf scale='min(320\,iw):-2' -movflags +faststart -metadata description=$KEY_PREFIX.mp4 out.mp4 -vf thumbnail -vf scale='min(320\,iw):-2' -vframes 1 out.png
```

I am unsure if I'm missing a critical configuration somewhere that prevents me from running it, could you please provide me with some guidance as to what might I be missing?

Thanks in advance :)

**Comment by binoculars** [comment]
Best guess is that you need to single quote those environment variables like in the README. The FFmpeg command appears to be exiting early without an error, so that's why I think that is the problem. If there was an error, it would be logged. Also,

**Comment by yabrego** [comment]
Thanks for the suggestion, it unfortunately didn't work, I was however able to make the FFMPEG run by using -loglevel debug at the beginning but it still doesn't work properly.

I tried with a bunch of different combinations on the quote for VIDEO_MAX_DURATION, MIME_TYPES and FFMPEG_ARGS, none of which worked, to clarify those are the values as the are on the lambda console (not my terminal).

So after the changes, and different tests, it looks like this now, and it runs FFMPEG
```
FFMPEG_ARGS=-loglevel debug -c:a copy -vf scale='min(320\,iw):-2' -movflags +faststart -metadata description=$KEY_PREFIX.mp4 out.mp4 -vf thumbnail -vf scale='min(320\,iw):-2' -vframes 1 out.png
```

It did however time out on a rather small video (3seconds HD)

So I feel like I'm still doing something wrong, furthermore, that didn't even produce FFMPEG logs in cloudwatch.

Is there something I can modify in the code to retrieve ffmpeg logs and print them?

**Comment by binoculars** [comment]
The only way to get the logs is in cloudwatch. It's not producing a log, so there's nothing to get. Maybe you can post the sample video you're using and I can try it?

This is what my console has:
<img width="1610" alt="screenshot" src="https://cloud.githubusercontent.com/assets/3374510/26500816/5dcb6118-4205-11e7-9516-73489d77b98b.png">

**Comment by binoculars** [comment]
Thinking about it more, it might be that the KEY_PREFIX isn't properly escaped. That could be a bug. Maybe try a key without slashes, e.g. 'abc.mp4' vs 'path/to/abc.mp4' in your s3 bucket. If that's the case, wrapping the description in single quotes, or deleting it and the metadata flag altogether,  might be the solution.

**Comment by yabrego** [comment]
I managed to get a log, I modified the ffmpeg function in commons.js to redirect the stdout of the spawn of ffmpeg to the console, it works now, but the output file has 0 bytes.
[test.zip](https://github.com/binoculars/aws-lambda-ffmpeg/files/1032419/test.zip)
Had to zip the file to be able to upload it.

It does seem to be something about that escaping, the same file with brackets in the name produced a correct output.
